### ISSUE TITLE
Stage3 - Permission errors on certain copies

### ISSUE THREAD

**Issue by roleymoley** [konveyor/pvc-migrate#137]
Trying 1.1.2 we get partial success during some test migrations. One issue we have seen here though is in some circumstances the rsync fails with errors such as

``` 
<d-worker08.in.appcanvas.net> Failed to connect to the host via ssh: Shared connection to d-wor
ker08.in.appcanvas.net closed.
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
<d-worker08.in.appcanvas.net> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o S
trictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-w
ith-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o C
ontrolPath=/home/ec2-user/.ansible/cp/811f235127 d-worker08.in.appcanvas.net '/bin/sh -c '"'"'r
m -f -r /home/ec2-user/.ansible/tmp/ansible-tmp-1599815055.085095-17521-23763704473814/ > /dev/
null 2>&1 && sleep 0'"'"''
<d-worker08.in.appcanvas.net> (0, b'', b'')
fatal: [localhost -> d-worker08.in.appcanvas.net]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
"cmd": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.volumes/pods/ed4
4077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-131e-11e9-9098-0
a15905bf50e/ rsync://root@localhost/postgresql/",
    "delta": "0:00:00.025510",
    "end": "2020-09-11 14:34:15.496229",
    "invocation": {
        "module_args": {
            "_raw_params": "rsync -aPvvHh --delete --port 2222 /var/lib/origin/openshift.local.
volumes/pods/ed44077c-efc6-11ea-afc3-0a935e4bbdfa/volumes/kubernetes.io~glusterfs/pvc-f9a8f858-
131e-11e9-9098-0a15905bf50e/ rsync://root@localhost/postgresql/",
            "_uses_shell": true,
            "argv": null,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "stdin_add_newline": true,
            "strip_empty_ends": true,
            "warn": true
        }
    },
    "msg": "non-zero return code",
    "rc": 5,
    "start": "2020-09-11 14:34:15.470719",
    "stderr": "@ERROR: access denied to postgresql from localhost (127.0.0.1)\nrsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]",
    "stderr_lines": [
        "@ERROR: access denied to postgresql from localhost (127.0.0.1)",
        "rsync error: error starting client-server protocol (code 5) at main.c(1649) [sender=3.1.2]"
    ],
    "stdout": "opening tcp connection to localhost port 2222\nsending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . postgresql/  (6 args)",
    "stdout_lines": [
        "opening tcp connection to localhost port 2222",
        "sending daemon args: --server -vvlHogDtpre.iLsfxC --delete --partial . postgresql/  (6 args)"
    ]
}
Read vars_file 'vars/run-rsync.yml'
Read vars_file 'vars/defaults.yml'
Read vars_file 'vars/run-rsync.yml'
Read vars_file 'vars/defaults.yml'
```

This is only for some PVCs so probably something specific to postgres pods from the looks of the information presented.

**Comment by jwmatthews** [comment]
@pranavgaikwad @alaypatel07 please take a look

**Comment by pranavgaikwad** [comment]
@roleymoley #136 is the same issue as this. Can you please confirm that the rsync task is running as root by following https://github.com/konveyor/pvc-migrate/issues/136#issuecomment-686620812?

In the output you posted, it doesn't look like rsync is running as root user. Another line in the output that confuses me is: 
```sh
<d-worker08.in.appcanvas.net> ESTABLISH SSH CONNECTION FOR USER: None
```
Why is the user None?

**Comment by pranavgaikwad** [comment]
Reopening as RC was wrong

**Comment by pranavgaikwad** [comment]
The root cause of the issue is identified to be a missing hosts config in rsyncd configuration:

https://github.com/konveyor/pvc-migrate/blob/c1cf0dc544a370ae5d9a0da3c154b06ad142ddc6/3_run_rsync/templates/rsyncd.yml.j2#L30

The stunnel service makes it look like the remote connection is originating from `localhost`. In the original allow config, we only considered `::1` loopback as allowed. However, we should have also considered `localhost` loopback. 

FIxed through #141
